src/lib/reasoners/bitv.ml

```diff
@@ -1103,13 +1101,15 @@ module Shostak(X : ALIEN) = struct
       in
       slice_vars_fix pat [] [] eqs
 
-
     let apply_subs subs sys =
       let rec f_aux = function
-        |[] -> []
-        |v::r ->
+        | [] -> []
+        | v::r ->
           match assoc_var v subs with
-          | vl -> vl @ f_aux r
+          | vl ->
+            (* Substitutions may be recursive and need to be applied again to
+               their result. *)
+            f_aux (vl @ r)
```

Comment:
We could, but given that C variables occur exactly twice (and one occurrence was eagerly substituted when the substitution was created) I don't think it would bring much benefit.